src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -284,6 +284,15 @@ ucs_status_t uct_rdmacm_cm_ep_disconnect(uct_ep_h ep, unsigned flags)
 
     ucs_trace("rdma_disconnect on cm_id %p", cep->id);
 
+    if (!ucs_unlikely(cep->flags & UCT_RDMACM_CM_EP_CONNECTED)) {
+        ucs_error("rdmacm_cm ep %p (id=%p) failed to disconnect from peer %p."
+                  "disconnect was called on an ep that is not connected",
+                  cep, cep->id,
+                  ucs_sockaddr_str(rdma_get_peer_addr(cep->id), ip_port_str,
+                                   UCS_SOCKADDR_STRING_LEN));
+        return UCS_ERR_IO_ERROR;
```

Comment:
Let's make it clearer.
I see 2 cases when it's possible:
1) good flow when peer is responded yet or it's on wire/queue, so need to return UCS_ERR_BUSY until connect callback will be invoked and state will be transited to "connected".
2) err flow when connect callback was invoked with rejected or err status, need to return UCS_ERR_INVALID_PARAM.
something else?

Reply:
I don't think this PR achieves its purpose.
@alinask can you pls add unit test which calls uct_ep_disconnect() twice and shows the 2nd time returns an error?

Reply:
i'll add a test.
the purpose was to fail and notify the use if he is calling uct_ep_disconnect on an ep that isn't connected. what should be different here?

Reply:
uct_ep_disconnect() should somehow update the state of the endpoint, so subsequent call to disconnect will fail with INVALID_PARAM

Reply:
IMHO, calling `ucp_ep_disconnect` twice should return `UCS_ERR_INVALID_PARAM` as well

Reply:
the ep is really considered disconnected when its disconnect callback is invoked.
the check in this pr should cover a case where uct_ep_disconnect is called before the ep even got a change to connect and if it was already disconnected. 
this pr prints an error if uct_ep_disconnect is called on an ep that isn't connected, calling this function more than once is a case of it. you think it should be handled separately?

Reply:
but actually, if uct_ep_disconnect is called "quickly" twice, i.e. the second call happens before the disconnect callback is invoked (due to the first call) then rdma_disconnect would be called twice on the same rdmacm id, which is bad.
i'll add a flag to mark the disconnect call on the ep, so that the ep state (connected or not) won't be considered.